src/main/java/edu/umich/its/spe/SPEUtils.java

```diff
@@ -19,8 +49,110 @@ static protected String safeGetPropertyValue(HashMap<String, String> testpropert
 
 	static String getISO8601StringForDate(Date date) {
 		// Get the current time (with time zone).
-		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss z", Locale.getDefault());
+		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
 		return dateFormat.format(date);
 	}
 
+	// Take out minor variations in a timestamp string.  This makes parsing
+	// much, much easier.
+
+	public static String normalizeStringTimestamp(String stringTimestamp) {
+
+		M_log.debug("normalizeStringTimestamp: input: {}",stringTimestamp);
+
+		if (stringTimestamp == null || stringTimestamp.length() == 0) {
+			return null;
+		}
+
+		//////// eliminate any fractional seconds.  Note this truncates, it does NOT round.
```

Comment:
What is the real reason for truncating second? purpose and will it affect the next SPE runtime( cron run I mean)

Reply:
Fractional seconds are just clutter and aren't significant.  Not all timestamps would contain fractional seconds so it is simpler to just get rid of them.  This won't impact the next SPE run in any meaningful way.